src/agents/parking_agent.py

```diff
@@ -154,6 +153,13 @@ async def run(self):
                 else:
                     reservation_status = self.agent.try_to_reserve_parking_spot(request.time_start, request.time_stop)
                     if reservation_status:
+                        new_reservation_id = (
+                            self.agent.generate_reservation_id(request.user_id) if reservation_status else ""
```

Comment:
Does this function actually need the self? Maybe we could make it static? Or move it out from the class?